### ISSUE TITLE
Can we commit / track package-lock.json?

### ISSUE THREAD

**Issue by dellagustin** [Podcastindex-org/web-ui#219]
Hello @daveajones , 

Can we commit `package-lock.json`? Without it the build is a bit flaky and can break when new versions of the dependencies are released, even when there is no code change in the project itself.

**Comment by dellagustin** [comment]
I think I just had one of those issues. `npm run dev` started failing due to missing module `dotenv-webpack` after I deleted `package-lock.json` and the `node_modules` and run `npm install` for a fresh start.
This module is not on `package.json` and this file was only changed the last time 6 months ago.
I can only imagine this was due to some transient dependency that changed over time because `package-lock.json`.

**Comment by stevencrader** [comment]
We have been using `yarn` for this project so there is a `yarn.lock` file already committed. You usually don't want both lock file types.

**Comment by dellagustin** [comment]
hmmm ok, that does make sense.

**Comment by dellagustin** [comment]
Follow up question, I installed yarn and typed `yarn` just to test if it was working, it triggered something that looks like an install process.
As result, the file `.yarnrc,yml` and folder `.yarn/cache` (with a bunch of zip files) were created.
I'm assuming that `.yarn/cache` has to go onto `.gitignore`, but how about `.yarnrc,yml`?

**Comment by stevencrader** [comment]
Did you install yarn local or global? I don't see those in my folder when `yarn` is installed globally. Regardless, I don't see any issue adding them to the ignore file. I don't see any reason we would want to track that information.

**Comment by stevencrader** [comment]
What version of `yarn` are you running? I'm running 1.22.15 on Windows.

**Comment by dellagustin** [comment]
@stevencrader it is installed globally. I recently installed it.
`yarn --version` spits out 3.4.1.
The content of `.yarnrc.yml`  is simply:
```
nodeLinker: node-modules
```
https://yarnpkg.com/configuration/yarnrc#nodeLinker
Maybe this is only required for newer versions?

As for committing `.yarn/cache`, I did some quick research. 
It depends on whether we want to follow the [Zero-install](https://yarnpkg.com/features/zero-installs) philosophy (essentially not needing to run `yarn install` after cloning, plus this is supposed to give you more stable builds in comparison to having to run `yarn install`). The drawback would be that it makes the tracked source code heavier.

By default we are not following this philosophy, so I would follow the [guidance from yarn's Q&A with respect to the `.gitignore`](https://yarnpkg.com/getting-started/qa#which-files-should-be-gitignored) and introduce the following:
```
.pnp.*
.yarn/*
!.yarn/patches
!.yarn/plugins
!.yarn/releases
!.yarn/sdks
!.yarn/versions
```
